lib/interfaces/ws.js

```diff
@@ -177,24 +177,31 @@ module.exports = function (app) {
             }
           })
         }
+
         if (msg.subscribe) {
-          app.subscriptionmanager.subscribe(
-            msg,
-            unsubscribes,
-            spark.write.bind(this),
-            app.securityStrategy.shouldFilterDeltas()
-              ? msg => {
-                var filtered = app.securityStrategy.filterReadDelta(
-                  spark.request,
-                  msg
-                )
-                if (filtered) {
-                  spark.write(filtered)
+          if (spark.query.startTime) {
+            if (app.historyProvider) {
+              app.historyProvider.addSubscription(spark, msg)
```

Comment:
Have you implemented `addSubscription` in Influxdb implementation? If a history provider does not support it how should we proceed? We have not defined any response mechanism for subscription protocol.

Reply:
Add subscription is not yet implemented. Did not know what to do without any response mechanism. Maybe need to close the stream so that the client knows something went wrong?

Reply:
Heh, I wrote the same suggestion but deleted it because it seemed too brutal. Then again, failing fast and refusing to fulfill the request possibly in a wrong way seems somewhat reasonable.